sdk/python/feast/big_query_source.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2019 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Optional
+
+
+class BigQuerySource:
+    def __init__(
+        self,
+        table_ref: Optional[str],
+        event_timestamp_column: str,
+        created_timestamp_column: Optional[str],
+        field_mapping: Optional[Dict[str, str]],
+        query: Optional[str],
+    ):
+        assert (table_ref is None) != (query is None), "Exactly one of table_ref and query should be specified"
```

Comment:
Small nitpick, but generally I think we should avoid using assert for errors that we expect users will make in lieu of normal exceptions

https://stackoverflow.com/questions/40182944/whats-the-difference-between-raise-try-and-assert#:~:text=raise%20is%20typically%20used%20when,if%20a%20condition%20is%20met.&text=If%20production%20quality%20code%20raises,ve%20got%20a%20bigger%20problem.

Assert should normally be used for impossible to reach situations.